domain/optatask/src/main/java/org/optatask/domain/Task.java

```diff
@@ -48,12 +48,14 @@
     // Task nextTask inherited from superclass
     @AnchorShadowVariable(sourceVariableName = "previousTaskOrEmployee")
     private Employee employee;
-    @CustomShadowVariable(variableListenerClass = StartTimeUpdatingVariableListener.class,
+    @CustomShadowVariable(variableListenerClass = StartAndEndTimeUpdatingVariableListener.class,
             // Arguable, to adhere to API specs (although this works), nextTask and employee should also be a source,
             // because this shadow must be triggered after nextTask and employee (but there is no need to be triggered by those)
             sources = {@PlanningVariableReference(variableName = "previousTaskOrEmployee")})
     private Integer startTime; // In minutes
-    private Integer endTime;
+    @CustomShadowVariable(variableListenerClass = StartAndEndTimeUpdatingVariableListener.class,
+            sources = {@PlanningVariableReference(variableName = "previousTaskOrEmployee")})
```

Comment:
Replace this with a @CustomShadowVariable(reference to "startTime"), so it piggybacks.
Now the variable listener runs twice for all changes and each one updates both shadow variables, so it's duplicate work.